### DISCUSSION TITLE
Copy open tabs to clipboard to paste in Zotero note

### DISCUSSION THREAD

**Discussion by mjthoraval** [windingwind/zotero-actions-tags#208] [Q&A] [answered]
I would like to copy the open tabs to a format that I can paste into a Zotero note.
The final format I would like to get is (in a Zotero Note Bulleted List, with active links from a click on the last 2 [links]):
- [Quick Copy formatted citation] [Select link] [Open-pdf link]
- [Quick Copy formatted citation] [Select link] [Open-pdf link]
- ...

In this way, I can simply save to a note all the open tabs, close all the tabs, annotate in the note and reopen the tabs one by one as needed.
Navigating through the open tabs is too slow, so getting the citation information with links to a note would be very useful.

It is already possible to do something similar to export to Markdown: https://github.com/windingwind/zotero-actions-tags/discussions/206
But it only shows text when exported to a Zotero note, without active links.
I would prefer staying in Zotero rather than going to another software.

It is also possible to save and restore tabs from this one: https://github.com/windingwind/zotero-actions-tags/discussions/129
But it can only reopen all the tabs, which is too slow when I just want to access the different papers one by one.

**Comment by s1291** [comment]
Use this script: https://github.com/windingwind/zotero-actions-tags/discussions/206

When you paste in a Zotero note, it doesn't recognize the Markdown URLs immediately for some reason, but if you delete the last `)` on each line, then retype it will will recognize the URLs.

**Reply by windingwind** [reply]
The clipboard helper supports copy in multiple formats at the same time. Use HTML for Zotero notes.

**Reply by s1291** [reply]
Zotero notes do not appear to support HTML. I have pasted some HTML URLs but they appear as plain text.

**Reply by windingwind** [reply]
``clipboard.addText(`<a href="${uri}">${linkText}</a>`, "text/html");`` from #115 

You'll need to copy as "text/html".

**Reply by s1291** [reply]
Why would you need to copy text as HTML? in my first reply I mentioned that Markdown already works on Zotero notes. HTML is not rendered in Zotero notes as far as I know (e.g. I have tested that on Zotero 7 beta). 

Sorry if I am missing something important in your replies.

**Reply by windingwind** [reply]
I mean, if you copy the HTML text as "text/html", then when you paste to the Zotero note editor, it will be correctly rendered.

The markdown as "text/unicode"certainly can be translated to rich-text by note editor, but that requires an input or deletion to trigger.

Open console (menu -> Tools -> Developer -> Error Console), run this line, and paste to a Zotero note, you'll see it be correctly rendered.

```js
new Zotero.ActionsTags.api.utils.ClipboardHelper().addText('<a href="https://zotero.org/">Zotero</a>', 'text/html').copy()
```

**Reply by s1291** [reply]
Thank you for providing that example. Indeed, that works as expected.

**Reply by windingwind** [reply]
You can add both markdown as plain text and HTML as HTML to a ClipboardHelper instance, which allows you to copy both and paste as MD in MD editor and as HTML in rich-text editor. (Technically, you can even also copy file and image to clipboard all at once and this will allow you to copy-paste file and image as well)

**Reply by s1291** [reply, marked as answer]
@windingwind : Many thanks for the suggestions. I have updated the script here: https://github.com/windingwind/zotero-actions-tags/discussions/206

@mjthoraval : See the updated script.

**Reply by mjthoraval** [reply]
Thank you very much!
